posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,311 @@
+
+* Test applications;
+* The actual tests;
+* Server configuration files used by the tests;
+
+In a similar fashion, application developers can use the Eclipse Transformer to transform their applications and test artifacts.
+
+The transformer is quick and efficient.  When integrated as a build step, as is done in Open Liberty builds, use of the transformer avoids the creation of new Jakarta EE 9 source files.
+
+== Transformation Strategy
+
+The stratgy used for Jakarta EE 9 test enablement in Open Liberty builds is to annotate test classes and methods to indicate which of these are to be run as Jakarta EE 9 tests.  As an added step, the test framework transforms and re-runs the annotated tests as Jakarta EE 9 tests.
+
+Two specific updates are required when transforming test artifacts.  First, package references are updated, changing Java / Jakarta EE "javax" package references into Jakarta EE 9 "jakarta" package references.  Package references can be in several locations, including test applications, test code, and server configurations.  Second, server configurations are updated to replace Java / Jakarta EE features with corresponding Jakarta EE 9 features.  The Eclipse Transformer is used to perform the first of these updates.  Currently, updates to server configurations must be done manually.  (Server configuration features could be updated using the transformer, if rules data were created with the substitution values needed for this update.)
+
+== Invoking the Transformer
+
+The Eclipse Transformer can be used directly from the command line, as a build task, or from Java code.  This article describes using the transformer from the command line.
+
+(See the transformer project pages for detailed usage guidelines link:https://projects.eclipse.org/projects/technology.transformer[Eclipse Transformer].)
+
+=== Downloading the Transformer 
+
+Several downloads are available from the project pages link:https://projects.eclipse.org/projects/technology.transformer/downloads[Downloads].
+
+Current snapshots are available at link:https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/org.eclipse.transformer.cli/[Snapshots].
+
+=== Building the transformer
+
+Alternatively, the transformer can be built by cloning the link:https://github.com/eclipse/transformer[Eclipse Transformer repository] and invoking maven to build the transformer jar:
+
+[source]
+--
+    maven clean package
+--
+
+=== Preparing a work area
+
+Download or copy the transformer binaries to a local directory:
+
+[source]
+--
+    mkdir <work>
+    cd <work>
+    cp transformer/org.eclipse.transformer.cli/target/org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar .
+    unzip org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar
+--
+
+Unpackaging the distribution obtains:
```

Comment:
This wording sounds weird.  Maybe something like:  `After unzipping, the directory contains:`

Reply:
Yeah, I was wondering about this, too.  I changed it to "unpacking".

Reply:
It still sounds weird